File: src/main/scala/ru/tolsi/lykke/waves/api/wallet/SignService.scala
package ru.tolsi.lykke.waves.api.wallet

import com.wavesplatform.wavesj.PrivateKeyAccount
import com.wavesplatform.wavesj.Transaction
import ru.tolsi.lykke.waves.wallet.UnsignedTransferTransaction
import scala.concurrent.Future

trait SignService {
  def createAccount(): Future[PrivateKeyAccount]
  def sign(utx: UnsignedTransferTransaction): Future[Transaction]
}

File: src/main/java/ru/tolsi/lykke/waves/wallet/SignService.java
package ru.tolsi.lykke.waves.wallet;

import com.wavesplatform.wavesj.PrivateKeyAccount;
import com.wavesplatform.wavesj.Transaction;

public interface SignService {
    PrivateKeyAccount createAccount();
    Transaction sign(UnsignedTransferTransaction utx);
}
